main.py

```diff
@@ -206,6 +214,7 @@ def main(collect: False, resume: Path = None):
         parameters['uploads'] = FileManager.file_paths_to_dict(resume, plain_text_resume_file)
         parameters['outputFileDirectory'] = output_folder
         parameters['collectMode'] = collect
+        parameters['browser'] = browser
```

Comment:
take care or too much movement of this variable. use singleton state for storing this and use them whenever you require, not passing browser in control flow and methods (also consider transitive changes))

factroy method / factroy patten would solve this

Reply:
Thanks for the suggestions, I've refactored it to use BrowserFactory to set the browser type based on the value specified in the config file.